src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -500,6 +500,14 @@ event::ptr primitive_inst::realloc_if_needed() {
         }
     }
 
+    if (_node->is_type<crop>()) {
+        if (can_be_optimized()) {
+            this->_outputs[0] = dep_memory_ptr(0);
```

Comment:
Why dont we apply this in on_execute() => update_output_memory() ?

Reply:
updated to set output memory in `crop_inst::update_output_memory` and just return event here to skip `realloc_if_needed`